### PR TITLE
feat(checkout): update order interface

### PR DESCRIPTION
**Pull request by griest024**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/graycoreio/daffodil/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Fixes: N/A

## What is the new behavior?
Changes the order service interface:
- Removes `placeOrder`.
- Adds `get`
- Adds `list`

## Does this PR introduce a breaking change?
```
[x] Yes
[ ] No
```

Prefixes order models with `Daff`.

## Other information

### CODE REVIEW

File: libs/checkout/src/drivers/interfaces/checkout-service.interface.ts [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import { Observable } from 'rxjs';
 
-import { Order } from '../../models/order/order';
+import { DaffOrder } from '../../models/order/order';
 
 export interface DaffCheckoutServiceInterface {
```
Comment (lderrickable) [inline libs/checkout/src/drivers/interfaces/checkout-service.interface.ts:5]: shouldn't this be deprecated?

Comment (griest024) [reply libs/checkout/src/drivers/interfaces/checkout-service.interface.ts:5]: Yeah it should be deprecated. There's a lot of stuff in the checkout package that needs to be changed.

**Comment by lderrickable** [comment]
The description of this PR says that placeOrder is removed, but it isn't, right?

**Comment by lderrickable** [comment]
oh, nevermind, I see

**Comment by lderrickable** [comment]
lgtm

**Pull request merged**